src/hotspot/share/cds/dynamicArchive.cpp

```diff
@@ -410,7 +410,8 @@ void DynamicArchive::dump(const char* archive_name, TRAPS) {
 }
 
 bool DynamicArchive::should_dump_at_vm_exit() {
-  return DynamicDumpSharedSpaces && (ArchiveClassesAtExit != nullptr);
+  return (DynamicDumpSharedSpaces && (ArchiveClassesAtExit != nullptr)) ||
+         (AutoCreateSharedArchive && Arguments::GetSharedDynamicArchivePath() != nullptr);
```

Comment:
I think it's better to do this: if `AutoCreateSharedArchive` is true, and you decide that the archive needs to be updated, then:

```
    DynamicDumpSharedSpaces = true;
    ArchiveClassesAtExit = Arguments::GetSharedDynamicArchivePath();
```

That way we don't need to do the check at line 414 anymore.